### DISCUSSION TITLE
Spindle to encoder gear ratio - does it have to be 1:1 ?

### DISCUSSION THREAD

**Discussion by EdFleta** [kachurovskiy/nanoels#71] [Q&A]
Hi Maxim,
Congratulations again for your excellent job developping Nano ELS.
I have followed your instructions and bought a 600 step encoder. However I am not using the gear set up and would prefer a GT2 belt system. 

As asked in the title is it possible to use different pully diameters e.g 100teeth spindle/80 teeth encoder and what would be the impact on the code ?
best regards, Eddie

**Comment by kachurovskiy** [comment]
Hi Eddie, you'd need to adjust `ENCODER_STEPS` accordingly in https://github.com/kachurovskiy/nanoels/blob/main/h1/NanoEls.ino#L6 - for 100t spindle 80t 600 step encoder it would be 600*100/80=750

**Reply by EdFleta** [reply]
great. thanks a lot

eddie

**Reply by lilly1charlie1tim** [reply]
Hi all, I am going to try and build this project for a myford ml7, asking here as it seems related to me. 
Does the stepper motor when attached to the leadscrew need to have a gear ratio of 1:1 also, or can this be changed in the code also?

I was also thinking of mounting it with a 90 bevel gear. I am assuming direction not an issue.
thanks

**Reply by kachurovskiy** [reply]
> Does the stepper motor when attached to the leadscrew need to have a gear ratio of 1:1 also, or can this be changed in the code also?

It can have any ratio but it's best to have a non-fractional numbers of motor steps per 1 turn of the screw.

> I was also thinking of mounting it with a 90 bevel gear. I am assuming direction not an issue.

It'll probably be fine but consider using a belt instead since it's much quieter and doesn't rattle (change direction for a fraction of a second while spinning).

**Reply by lilly1charlie1tim** [reply]
Thanks for the info, superb work you have done and thanks for making it open source!
